core/src/test/groovy/com/novoda/gradle/release/TestGeneratePomTask.groovy

```diff
@@ -0,0 +1,140 @@
+
+        File pomFile = new File(testProjectDir.root, '/build/publications/maven/pom-default.xml')
+        def nodes = new XmlSlurper().parse(pomFile)
+        def dependencies = nodes.dependencies.dependency
+
+        assert dependencies.size() == 3
+        assert dependencies.find { dep -> dep.artifactId == "hello" && dep.scope == "compile" } != null
+        assert dependencies.find { dep -> dep.artifactId == "haha" && dep.scope == "compile" } != null
+        assert dependencies.find { dep -> dep.artifactId == "world" && dep.scope == "runtime" } != null
+    }
+
+    @Test
+    void testGeneratePomTaskForAndroidLibrary() {
+        File manifestFile = new File(testProjectDir.root, "/src/main/AndroidManifest.xml")
+        manifestFile.getParentFile().mkdirs()
+        manifestFile.createNewFile()
+        manifestFile << """
+            <manifest package="com.novoda.test"/>
+        """
+
+        buildFile << """
+            buildscript {
+                repositories {
+                    jcenter()
+                    google()
+                }
+                dependencies {
+                    classpath files($pluginClasspath)
+                    classpath 'com.android.tools.build:gradle:3.0.0'
+                }
+            }
+            apply plugin: "com.android.library"
+            apply plugin: "bintray-release"
+            
+            android {
+                compileSdkVersion 26
+                buildToolsVersion "26.0.2"
+
+                defaultConfig {
+                    minSdkVersion 16
+                    versionCode 1
+                    versionName "0.0.1"
+                }    
+            }
+            
+            publish {
+                userOrg = 'novoda'
+                groupId = 'com.novoda'
+                artifactId = 'test'
+                publishVersion = '1.0'
+            }
+        
+            dependencies {
+                compile 'com.abc:hello:1.0.0'
+                implementation 'com.xyz:world:2.0.0'
+                api 'com.xxx:haha:3.0.0'
+            }
+        """
+
+        def result = GradleRunner.create()
+            .withProjectDir(testProjectDir.root)
+            .withArguments("generatePomFileForReleasePublication")
+            .build()
+
+        assert result.task(":generatePomFileForReleasePublication").outcome == SUCCESS
+
+        File pomFile = new File(testProjectDir.root, '/build/publications/release/pom-default.xml')
+        def nodes = new XmlSlurper().parse(pomFile)
+        def dependencies = nodes.dependencies.dependency
+
+        assert dependencies.size() == 3
+        assert dependencies.find { dep -> dep.artifactId == "hello" && dep.scope == "compile" } != null
+        assert dependencies.find { dep -> dep.artifactId == "haha" && dep.scope == "compile" } != null
+        assert dependencies.find { dep -> dep.artifactId == "world" && dep.scope == "runtime" } != null
```

Comment:
Is `runtime` the correct one here? 🤔 
Have you a documentation you can refer to, I just don't know it :D

Reply:
https://developer.android.com/studio/build/gradle-plugin-3-0-0-migration.html#new_configurations

from the table, only `implementation` will be available at runtime, so `runtime` scope is the best match here. Generate pom with `java-library` also output the same.